[Microbyte] IPv4 Addressing

Anjalikamath opened the pull request:
## Closes #11 
Added a microbyte to learn some terminologies in IPv4 addressing
 Checklist: 
✅ Added activities for learners
✅ Didn't give the solutions to the added activities straight away 
✅ Followed a LearnByDoing approach 
✅ Added images for better understanding if deemed necessary

Anjalikamath committed:
Added IPv4 addressing

Anjalikamath committed:
Update README.md

Anjalikamath committed:
Update README.md

Anjalikamath committed:
Update README.md

YourAKShaw commented:
@Anjalikamath 

Kindly follow the contribution guidelines as mentioned in the [CONTRIBUTING.md](https://github.com/Crio-Bytes/Networking/blob/main/CONTRIBUTING.md) file of the repo. 

- According to our contribution guidelines, you are required to create an issue for a new MicroByte Proposal, similar to https://github.com/Crio-Bytes/Miscellaneous/issues/3. 
- You are required to utilize the `Add MicroByte` issue template for the same. 
- You need not wait for yourself to be assigned for your created issue, and can start working right away.
- Once, your work is done, you need to create a PR similar to https://github.com/Crio-Bytes/Miscellaneous/pull/1.
- You are requested to utilize [this](https://github.com/Crio-Bytes/Networking/blob/main/.github/PULL_REQUEST_TEMPLATE/adding-microbyte.md) PR template for the same.
- For further clarification, you can also go ahead and watch our YT video: [OSS Contribution Workflow of microbytes](https://www.youtube.com/watch?v=qdSn8v9nU0g&feature=youtu.be).

YourAKShaw commented:
Kindly create the required issue and edit your PR description.

Also, your PR is currently under review.

amoghrajesh commented:
Hey @Anjalikamath!
You need to open an issue pertaining to this PR by following an issue template as provided in .github folder of each repo. 
Post that, you need to create this particular PR details using a PR template provided in the same folder.
We will have a look asap after an issue is created and the PR details follow a template too 😄

Anjalikamath commented:
I am so sorry. I'll do the changes.

Anjalikamath commented:
Closes #11 
Added a microbyte to learn some terminologies in IPv4 addressing
 Checklist: 
✅ Added activities for learners
✅ Didn't give the solutions to the added activities straight away 
✅ Followed a LearnByDoing approach 
✅ Added images for better understanding if deemed necessary

YourAKShaw commented:
You needn't change the title. You had to edit the description. Do it similar to https://github.com/Crio-Bytes/Miscellaneous/pull/1.

If you have queries. Kindly reach out to us.

YourAKShaw commented:
> I am so sorry. I'll do the changes.

Chill. We're all here to learn.

Kevinpaulose05 commented:
@Anjalikamath Kindly remove ur solutions from the solution folder from ur microbyte. The reason is after we merge ur MB, u will be **opening issues** for learners to contribute to ur MB. The only way they can do is by answering ur quizzes. So u can open 5 issues and people can contribute into the solutions folder. U don't need to provide them. Else ppl will just use ur MB as a learning material and not as a contributing material.

Anjalikamath commented:
@Ak-Shaw  I'll do the changes.

Anjalikamath commented:
@Kevinpaulose05  Okay, I'll update that.

Anjalikamath committed:
Update README.md

Anjalikamath committed:
Update README.md

Anjalikamath committed:
Updated IPv4 addressing

Anjalikamath committed:
Merge branch 'main' of https://github.com/Anjalikamath/Networking into main

Anjalikamath committed:
Update README.md

Anjalikamath committed:
Update README.md

Anjalikamath committed:
Update README.md

Anjalikamath commented:
I have updated the microbyte. Kindly let me know if any changes are required.Thanks!

YourAKShaw commented:
@Anjalikamath I think that you're not getting it. :sweat_smile: 
![image](https://user-images.githubusercontent.com/51538194/97108918-855a8500-16f6-11eb-82ad-ea479e9d96b4.png)
You need to change the description above and use the aforementioned PR template for it.

Kevinpaulose05 commented:
@Anjalikamath Can u pls mention how learners can give the solutions to the quizzes?

My suggestion is:
- Make a demo folder with a demo file called solutions folder like before and indicate abt the folder in the readme itself. And also mention abt this in ur issues.

But this is just a suggestion. feel free to come up with ur own solution. The main idea is to let ppl know how & where they can give the solutions, as that is the most critical aspect of your microbyte.

Anjalikamath committed:
Changed folder structure

Anjalikamath committed:
Merge branch 'main' of https://github.com/Anjalikamath/Networking into main

Anjalikamath commented:
> @Anjalikamath Can u pls mention how learners can give the solutions to the quizzes?
> 
> My suggestion is:
> 
> * Make a demo folder with a demo file called solutions folder like before and indicate abt the folder in the readme itself. And also mention abt this in ur issues.
> 
> But this is just a suggestion. feel free to come up with ur own solution. The main idea is to let ppl know how & where they can give the solutions, as that is the most critical aspect of your microbyte.

I have changed the folder structure and I have mentioned about this in the Readme as well. Can you please tell me how I can mention about this in my issues? Thanks!

Kevinpaulose05 commented:
> > @Anjalikamath Can u pls mention how learners can give the solutions to the quizzes?
> > My suggestion is:
> > 
> > * Make a demo folder with a demo file called solutions folder like before and indicate abt the folder in the readme itself. And also mention abt this in ur issues.
> > 
> > But this is just a suggestion. feel free to come up with ur own solution. The main idea is to let ppl know how & where they can give the solutions, as that is the most critical aspect of your microbyte.
> 
> I have changed the folder structure and I have mentioned about this in the Readme as well. Can you please tell me how I can mention about this in my issues? Thanks!

Add ur solutions folder that u had initially. Inside it create folders for quizzes like quiz-1,2,3,.. Keep demo files in each(so that while pushing github recognisies the folder). Then wherever u have mentioned "take this quiz", mention "and subkit the solution here" where here points to quiz's solution folder. Then u don't have to mention that seperately in the issue. 

Hope this cleared ur doubt

Anjalikamath committed:
Changed folder structure

Anjalikamath committed:
Update README.md

Anjalikamath committed:
Update README.md

Anjalikamath committed:
Added Solutions folder

Anjalikamath committed:
Added Solutions folder

Anjalikamath committed:
Merge branch 'main' of https://github.com/Anjalikamath/Networking into main

Anjalikamath commented:
> > > @Anjalikamath Can u pls mention how learners can give the solutions to the quizzes?
> > > My suggestion is:
> > > 
> > > * Make a demo folder with a demo file called solutions folder like before and indicate abt the folder in the readme itself. And also mention abt this in ur issues.
> > > 
> > > But this is just a suggestion. feel free to come up with ur own solution. The main idea is to let ppl know how & where they can give the solutions, as that is the most critical aspect of your microbyte.
> > 
> > 
> > I have changed the folder structure and I have mentioned about this in the Readme as well. Can you please tell me how I can mention about this in my issues? Thanks!
> 
> Add ur solutions folder that u had initially. Inside it create folders for quizzes like quiz-1,2,3,.. Keep demo files in each(so that while pushing github recognisies the folder). Then wherever u have mentioned "take this quiz", mention "and subkit the solution here" where here points to quiz's solution folder. Then u don't have to mention that seperately in the issue.
> 
> Hope this cleared ur doubt

Thank You so much! This cleared my doubts. I have updated the microbyte.

YourAKShaw commented on IPV4_Addressing/README.md:
```diff
@@ -0,0 +1,357 @@
+3. Dive deep into TCP/IP stack <br/>
+4. Learn about OSI stack and difference with TCP/IP stack <br/>
+and many many more !
+
```
@Anjalikamath I really appreciate your efforts. This MicroByte has some nice scope for learners. Just one request though. Can you kindly add a few references in the 4 points mentioned in the `FURTHER EXPLORATION` section above.

Other than that, it LGTM! :rocket: 

Let's see what the other reviewers have to say.

Anjalikamath replied:
I have modified the file. Thanks!

YourAKShaw reviewed (changes requested):
Kindly make the necessary changes and we'll move forward with reviewing.

Anjalikamath committed:
Added references

amoghrajesh reviewed (approved):
Hey @Anjalikamath!
This is one of the most engaging content I have seen in a while. 
Choosing such a theory related topic and making it seem so interesting is something extraordinary. 
Keep up the good work, no complaints from my end.
Make the changes suggested by @Ak-Shaw and think of some learner issues too
Hoping to see more such work from you 😄

Anjalikamath committed:
Update README.md

Anjalikamath committed:
Update README.md

Anjalikamath commented:
Thank you so much! @amoghrajesh

YourAKShaw reviewed (approved):
@Anjalikamath Good work. This MicroByte will be really fruitful for learners! :rocket: 

Thank you for your contributions. :heart_decoration:

Kevinpaulose05 commented:
Well done @Anjalikamath . I appreciate ur prompt and perfect responses. Keep up the work. Try coming up with even more microbytes in the following week.

Happy Contributing!!

YourAKShaw commented:
@Anjalikamath Kindly showcase your MicroByte in the **#microbyte-showcase** channel on slack using [this](https://github.com/Crio-Bytes/Demo-Repo/blob/master/Maintainer%20Resources/new-microbyte-showcase-template.md) template.

YourAKShaw commented:
@all-contributors kindly add @Anjalikamath for:

- [x] code
- [x] doc
- [x] ideas
- [x] content

YourAKShaw commented:
@all-contributors kindly add @amoghrajesh for review and eventOrganizing.
